signature/jws/jwt.go

```diff
@@ -0,0 +1,152 @@
+package jws
+
+import (
+	"crypto"
+	"crypto/x509"
+	"encoding/base64"
+	"errors"
+	"fmt"
+
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/notaryproject/notation-core-go/signature"
+)
+
+// signingMethod is the interface for jwt.SigingMethod with additional method to
+// access certificate chain after calling Sign().
+type signingMethod interface {
+	jwt.SigningMethod
+
+	// CertificateChain returns the certificate chain.
+	//
+	// It should be called after calling Sign().
+	CertificateChain() ([]*x509.Certificate, error)
+
+	// PrivateKey returns the private key.
+	PrivateKey() crypto.PrivateKey
+}
+
+// remoteSigningMethod wraps the remote signer to be a SigningMethod
+type remoteSigningMethod struct {
```

Comment:
Why do we need different interface for remote and local signer? Isn't local signer a special case of remote signer?

Reply:
Because the local signer includes the `certificate chain` and `private key`, but remote signer doesn't have them and can only access them after signing. Both `remoteSigningMethod` and `localSigningMethod` implement `signingMethod` interface to mitigate the difference between them.

```
type signingMethod interface {
	jwt.SigningMethod

	// CertificateChain returns the certificate chain.
	//
	// It should be called after calling Sign().
	CertificateChain() ([]*x509.Certificate, error)

	// PrivateKey returns the private key.
	PrivateKey() crypto.PrivateKey
}
```